boa/benches/lexer.rs

```diff
@@ -17,9 +17,10 @@ static EXPRESSION: &str = r#"
 fn expression_lexer(c: &mut Criterion) {
     c.bench_function("Expression (Lexer)", move |b| {
         b.iter(|| {
-            let mut lexer = Lexer::new(black_box(EXPRESSION));
+            let mut lexer = Lexer::new(black_box(EXPRESSION.as_bytes()));
 
-            lexer.lex()
+            // Goes through and lexes entire given string.
+            while lexer.next().expect("Failed to lex").is_some() {}
```

Comment:
we could have a method in lexer `collect` that would return `Result<Vec<Token>, LexError>`